sinatra-contrib/lib/sinatra/config_file.rb

```diff
@@ -131,11 +124,9 @@ def config_file(*paths)
             raise UnsupportedConfigType unless ['.yml', '.erb'].include?(File.extname(file))
             logger.info "loading config file '#{file}'" if logging? && respond_to?(:logger)
             document = ERB.new(IO.read(file)).result
-            yaml = config_for_env(YAML.load(document)) || {}
-            yaml.each_pair do |key, value|
-              for_env = config_for_env(value)
-              set key, for_env unless value and for_env.nil? and respond_to? key
-            end
+            yaml = YAML.load(document)
+            config = config_for_env(yaml)
+            config.each_pair { |key, value| set(key, value) }
```

Comment:
You shouldn't remove the conditional expressions because the following unintentional behavior will happen.
```yaml
# config.yml
development:
  raise_errors: "foo"
```

```ruby
require 'sinatra'
require 'sinatra/config_file'

config_file './config.yml'

get '/' do
  settings.raise_errors.to_s #=> expected "false", but got "foo"
end
```

I think that it isn't desirable to automatically overwrite `settings` defined by default by config file settings.

Reply:
@namusyaka Running this against both master and this branch I notice the default setting being overwritten. I don't think the original conditional defends against this behaviour. I can't actually think of a condition where the original would be met at all.

However, I can add this behaviour if you'd prefer?

Reply:
Hmm, makes sense.
It seems that the sample I showed was wrong.
However, we must grasp the condition that the original conditional expression was trying to protect.

Welcome if you can look it up.
Then, if the original conditional expression is meaningful, we have to add a test on that matter.

Reply:
I have the case where this would fail. If the value is being set to nil and it's set using this style, the conditional will be met and setting ignored:

```yaml
# config.yml
raise_errors:
  development: 

# app.rb
settings.raise_errors #=> expected nil, got false
```

However in both other styles the conditional is not met, and the default value is overwritten:

```yaml
# config.yml
raise_errors:

# app.rb
settings.raise_errors #=> expected nil, got nil



# config.yml
development:
  raise_errors:

# app.rb
settings.raise_errors #=> expected nil, got nil
```

If we're changing the behaviour anyway do we really want to stop people from overriding default values using the config file? It seems reasonable that it should be possible to do so.

If not we can exclude all cases or have it function exactly as it did before.

Reply:
It's no problem, I think.
I've reviewed this change to a certain extent, but I judged that it is almost no problem.